cli/auth.go

```diff
@@ -215,14 +257,20 @@ func (c *viamClient) ensureLoggedIn() error {
 		return errors.New("not logged in: run the following command to login:\n\tviam login")
 	}
 
-	if c.conf.Auth.isExpired() {
-		if !c.conf.Auth.canRefresh() {
+	authToken, ok := c.conf.Auth.(*token)
+	if ok && authToken.isExpired() {
+		if !authToken.canRefresh() {
 			utils.UncheckedError(c.logout())
 			return errors.New("token expired and cannot refresh")
 		}
 
+		currentToken, ok := c.conf.Auth.(*token)
+		if !ok {
+			return errors.New("config auth was not a token or api key")
```

Comment:
It doesn't look like you check whether `Auth` is an api key here. Also don't you already have `authToken` defined above? Why do you need another cast here?

Reply:
removed